docs/dev/reports/README.md

```diff
@@ -0,0 +1,469 @@
+A2(Admission report - ephemeral) --> B(Admission reports controller)
+A3(Admission report - ephemeral) --> B(Admission reports controller)
+A4(Admission report - ephemeral) --> B(Admission reports controller)
+B(Admission reports controller) --> AGG1(Aggregate ephemeral admission reports)
+AGG1(Aggregate ephemeral reports) -->|create/update| AGG2(Admission report - aggregated)
+B(Admission reports controller) -->|delete| D1(Admission report - ephemeral)
+B(Admission reports controller) -->|delete| D2(Admission report - ephemeral)
+B(Admission reports controller) -->|delete| D3(Admission report - ephemeral)
+B(Admission reports controller) -->|delete| D4(Admission report - ephemeral)
+```
+
+With that in mind, it's important to note that most admission reports should be short lived and exist only until they have been aggregated.
+Aggregated admission reports can be identified by the `audit.kyverno.io/report.aggregate` label.
+
+Only the aggregated report is expected to be long lived and evtually share the same lifecycle as the underlying resource.
+To aggergate reports together, Kyverno groups results per policy/rule and keeps the most recent one.
+
+### Backgroud scan reports
+
+Background scan reports are produced in the background at regular intervals or when something changes in the policy or the resource.
+There is always a one-to-one relationship between a resource and a background scan report.
+
+Kyverno updates the same background scan report over and over when policies are evaluated against a resource in the background scan process.
+
+```mermaid
+flowchart LR
+R(Resource changes) --> B(Background scan controller)
+P(Policy changes) --> B(Background scan controller)
+T(At regular intervals) --> B(Background scan controller)
+B(Background scan controller) --> E(Evaluates policies against existing resource)
+E(Evaluates policies against existing resource) -->|create/update| REPORT(Background scan report)
+```
+
+### Policy reports
+
+Policy reports (and cluster policy reports) are the result of aggregating admission reports and background scan reports per policy, at the namespace level (for namespaced resources) and at the cluster level (for cluster wide resources).
+
+Please note that only aggregated admission reports are considered, it would not make sense to consider short lived admission reports here.
+
+To reduce the computation power needed, especially when resources change frequently, we introduce a queue and a delay (`30s`):
+- Watch when an admission report and/or background scan report is CREATED, UPDATED or DELETED
+- Put the namespace of the report in the queue (`""` if the report is clustered) with a delay of `30s`
+- The queue guarantees that only a single worker will process an item at the same time
+- When the worker gets the namespace from the queue it lists all admission reports and background scan reports for that particular namespace, aggregates them together and reconciles the existing policy reports
```

Comment:
Once we create the per policy report, why do we still need to keep the per object report?

Reply:
To be able to reconcile the final report